backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java

```diff
@@ -143,28 +153,56 @@ public ResponseEntity createAssignment(
               );
 
       for ( PhoneNumberRequestEntity phoneNumberRequest : contactInfoRequest.getPhoneNumbers() ) {
-        PhoneNumberEntity phoneNumberEntity = PhoneNumberEntity
-            .builder()
-            .type( phoneNumberRequest.getType() )
-            .number( phoneNumberRequest.getNumber() )
-            .contactInfo( contactInfoEntityFromDB )
-            .build();
-        PhoneNumberEntity mergedPhoneNumber = entityManager.merge( phoneNumberEntity );
+        PhoneNumberEntity phoneNumberEntity = PhoneNumberMapper.mapToPhoneNumber( phoneNumberRequest );
+        phoneNumberEntity.setContactInfo( contactInfoEntityFromDB );
 
-        phoneNumberRepository.save( mergedPhoneNumber );
+        phoneNumberRepository.save( phoneNumberEntity );
       }
 
       for ( AddressRequestEntity addressRequest : contactInfoRequest.getAddresses() ) {
         AddressEntity addressEntity = AddressMapper.mapToAddress( addressRequest );
         addressEntity.setContactInfo( contactInfoEntityFromDB );
-        AddressEntity mergedAddress = entityManager.merge( addressEntity );
 
-        addressRepository.save( mergedAddress );
+        addressRepository.save( addressEntity );
       }
     }
 
     return ResponseEntity
         .ok()
         .body( assignmentEntityFromDB.getAssignmentId() );
   }
+
+  @Override
+  public ResponseEntity getAssignments() {
+    UserEntity user = ( UserEntity ) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    List<AssignmentResponseEntity> assignmentsResponse = new ArrayList<>();
+
+    for ( AssignmentEntity assignment : user.getAssignments() ) {

```

Comment:
umm why does user has assignments inside?
are you suggesting to extract all assignments on user information receival?
Please don't. It is better to have assignments list as separate Repository call. Imagine having 10+ users with 100+ assignments for each, seccurity User can be a cached entity so you will store 1000+ records in memory for your application for unknown period of time.

Reply:
Done: Removed assignments list from UserEntity, made request to DB to get assignments list